File: backend/src/domain/entities/MentorProfile.ts
import { Entity, PrimaryColumn, Column, CreateDateColumn } from 'typeorm';

@Entity('mentor_profiles')
export class MentorProfile {
  @PrimaryColumn({ name: 'member_id' })
  memberId: number;

  @Column({ type: 'varchar', name: 'current_job_title', nullable: true })
  currentJobTitle: string;

  @Column({ type: 'varchar', name: 'current_company', nullable: true })
  currentCompany: string;

  @Column({ type: 'text', nullable: true })
  bio: string;

  @Column({
    type: 'decimal',
    name: 'rating_avg',
    default: 0,
    precision: 10,
    scale: 2,
  })
  ratingAvg: number;

  @Column({ type: 'integer', name: 'total_sessions', default: 0 })
  totalSessions: number;

  @Column({ type: 'boolean', name: 'is_approved', default: false })
  isApproved: boolean;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend_new_structure/common/src/main/java/com/service/common/entity/MentorProfile.java
package com.service.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("mentor_profiles")
public class MentorProfile {

    @Id
    @Column("member_id")
    private Long memberId;

    @Column("current_job_title")
    private String currentJobTitle;

    @Column("current_company")
    private String currentCompany;

    private String bio;

    @Column("rating_avg")
    @Builder.Default
    private BigDecimal ratingAvg = BigDecimal.ZERO;

    @Column("total_sessions")
    @Builder.Default
    private Integer totalSessions = 0;

    @Column("is_approved")
    @Builder.Default
    private Boolean isApproved = false;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
